### PR TITLE
etcd-runner: refactor code structure and flag cleanup

### PR DESCRIPTION
**Pull request by xiang90**
replace #6735 

I got some time today to clean up runner code. I feel this is cleaner than 6735.

PTAL /cc @heyitsanthony @sinsharat

### CODE REVIEW

File: tools/functional-tester/etcd-runner/watcher.go [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+	gr = getWithRetry(client, ctx, "non-existant")
+	revision = gr.Header.Revision
+
+	ctxt, cancel := context.WithDeadline(ctx, time.Now().Add(runningTime))
+	defer cancel()
+
+	// generate and put keys in cluster
+	limiter := rate.NewLimiter(rate.Limit(reqRate), reqRate)
+
+	go func() {
```
Comment (xiang90) [inline tools/functional-tester/etcd-runner/watcher.go]: @heyitsanthony The code structure around this is still not ideal. I will clean this up later.

**Comment by xiang90** [comment]
@sinsharat Fixed.

File: tools/functional-tester/etcd-runner/watcher.go [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+)
+
+func runWatcher(getClient getClientFunc, limit int) {
+	ctx := context.Background()
+	for round := 0; round < limit; round++ {
+		performWatchOnPrefixes(ctx, getClient, round)
+	}
+}
+
+func performWatchOnPrefixes(ctx context.Context, getClient getClientFunc, round int) {
```
Comment (sinsharat) [inline tools/functional-tester/etcd-runner/watcher.go]: @xiang90 The othe issue which i feel with this code is that if we change etcd runner, to have flags for running time, noOfPrefixes, watcherPePrefix, reqRate the only way we can have it with the current design is to pass it as args (which will again make it complicated as will end up passing too many args to it) or to use these flags as global args which again will be a problem. Its just an example which i have mentioned which may or may not be applicable here but can happen in future for this or for other newly added etcd-runner feature. So probably making it exactly like etcdctl usage will be more appropriate as you had mentioned earlier.

Comment (xiang90) [inline tools/functional-tester/etcd-runner/watcher.go]: Right. If we need a lot of sub flags for watcher runner then we need subcommand: test watcher -x -y. And yes, we need to do what we have done in etcdctl: using a command line lib and put flags into command line context before passing through business logic func

**Comment by heyitsanthony** [comment]
lgtm

**Pull request merged**